backend/src/main/java/com/woowacourse/gongcheck/presentation/aop/HostVerifier.java

```diff
@@ -0,0 +1,27 @@
+package com.woowacourse.gongcheck.presentation.aop;
+
+import com.woowacourse.gongcheck.exception.UnauthorizedException;
+import com.woowacourse.gongcheck.presentation.AuthenticationContext;
+import com.woowacourse.gongcheck.presentation.Authority;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.springframework.stereotype.Component;
+
+@Aspect
+@Component
+public class HostVerifier {
+
+    private final AuthenticationContext authenticationContext;
+
+    public HostVerifier(final AuthenticationContext authenticationContext) {
+        this.authenticationContext = authenticationContext;
+    }
+
+    @Before("@annotation(com.woowacourse.gongcheck.presentation.aop.HostOnly)")
+    public void checkHost() {
+        final Authority authority = authenticationContext.getAuthority();
+        if (!authority.equals(Authority.HOST)) {
+            throw new UnauthorizedException("호스트만 입장 가능합니다.");
+        }
+    }
```

Comment:
```suggestion
        Authority authority = authenticationContext.getAuthority();
        if (!authority.isHost()) {
            throw new UnauthorizedException("호스트만 입장 가능합니다.");
        }
    }
```
메소드 필드에는 final을 붙이지 않습니다
또한 `Authority.Host`를 하드코딩하기보다는 authority 객체에 위임하면 더 좋을 것 같아요

지금처럼 `HostVerifier`에서 해당 로직을 태울지, authority 객체에 태울지, 혹은 `Authority`에서 static 메소드로 관리할지는 조금더 고민해보면 좋을 것 같아요
정답이있다기보다는 무엇이 좋은지 판단하고 변경하지 않아도 됩니다~

```java
Authority authority = authenticationContext.getAuthority();
authority.checkHostAuthority();
```
```
Authority.checkHostAuthority(authentication.Context);
```

Reply:
`AuthenticationContext` 는 요청 인증에 필요한 데이터를 머금고 있는 객체라고 생각합니다.
때문에 AOP 로 적용되는 `HostVerifier` 가 예외를 던지는 주체가 되어야 한다고 생각했습니다.
이에 따라서 `Authority` enum 에서는 값을 비교하는 메서드만 만들고, 검증 로직은 `HostVerifier` 에서 수행하도록 결정하였습니다.
```java
public void checkHost() {
    Authority authority = authenticationContext.getAuthority();
    if (!authority.isHost()) {
        throw new UnauthorizedException("호스트만 입장 가능합니다.");
    }
}
```